tracecat/expressions/functions.py

```diff
@@ -353,11 +407,142 @@ def to_timestamp_str(x: datetime) -> float:
     return x.timestamp()
 
 
+def extract_second(x: datetime) -> int:
+    return x.second
+
+
+def extract_minute(x: datetime) -> int:
+    return x.minute
+
+
+def extract_hour(x: datetime) -> int:
+    return x.hour
+
+
+def extract_day_of_week(x: datetime, format: str = "number") -> int | str:
```

Comment:
```suggestion
def extract_day_of_week(x: datetime, format: Literal["number", "full", "short"]  = "number") -> int | str:
```
and `from typing import Literal` at the top